FWCore/Framework/interface/EDConsumerBase.h

```diff
@@ -267,6 +294,59 @@ namespace edm {
     bool frozen_;
     bool containsCurrentProcessAlias_;
   };
+
+  template <Transition TR>
+  class EDConsumerBaseESAdaptor {
+  public:
+    template <typename TYPE, typename REC>
+    ESGetToken<TYPE, REC> consumes() const {
+      return m_consumer->template esConsumes<TYPE, REC, TR>();
+    }
+
+  private:
+    //only EDConsumerBase is allowed to make an instance of this class
+    friend class EDConsumerBase;
+    EDConsumerBaseESAdaptor(EDConsumerBase* iBase) : m_consumer(iBase) {}
+
+    EDConsumerBase* m_consumer;
+  };
+
+  template <Transition TR>
+  class EDConsumerBaseWithTagESAdaptor {
+  public:
+    template <typename TYPE, typename REC>
+    ESGetToken<TYPE, REC> consumes() const {
```

Comment:
```suggestion
    ESGetToken<TYPE, REC> consumes() {
```
assuming the function would better be non-`const`.